partiql-eval/src/eval/evaluable.rs

```diff
@@ -1339,8 +1339,203 @@ impl EvalExpr for EvalSubQueryExpr {
+#[derive(Debug, PartialEq)]
+pub(crate) struct EvalOuterUnion {
+    pub(crate) setq: SetQuantifier,
+    pub(crate) l_input: Option<Value>,
+    pub(crate) r_input: Option<Value>,
+}
+
+impl EvalOuterUnion {
+    pub(crate) fn new(setq: SetQuantifier) -> Self {
+        EvalOuterUnion {
+            setq,
+            l_input: None,
+            r_input: None,
+        }
+    }
+}
+
+impl Evaluable for EvalOuterUnion {
+    fn evaluate(&mut self, _ctx: &dyn EvalContext) -> Value {
+        let lhs = self.l_input.take().unwrap_or(Missing);
+        let rhs = self.r_input.take().unwrap_or(Missing);
+        match self.setq {
+            SetQuantifier::All => {
+                let mut result = coerce_to_bag(lhs);
+                coerce_to_bag(rhs).into_iter().for_each(|elem| {
+                    result.push(elem);
+                });
+                Value::from(result)
+            }
+            SetQuantifier::Distinct => {
+                let mut result: HashSet<Value> = HashSet::from_iter(coerce_to_bag(lhs).into_iter());
+                coerce_to_bag(rhs).into_iter().for_each(|elem| {
+                    result.insert(elem);
+                });
+                Value::from(Bag::from(result))
+            }
+        }
+    }
+
+    fn update_input(&mut self, input: Value, branch_num: u8, ctx: &dyn EvalContext) {
+        match branch_num {
+            0 => self.l_input = Some(input),
+            1 => self.r_input = Some(input),
+            _ => ctx.add_error(EvaluationError::IllegalState(
+                "Invalid branch number".to_string(),
+            )),
+        }
+    }
+}
+
+/// Represents the `OUTER INTERSECT` bag operator.
+#[derive(Debug, PartialEq)]
+pub(crate) struct EvalOuterIntersect {
+    pub(crate) setq: SetQuantifier,
+    pub(crate) l_input: Option<Value>,
+    pub(crate) r_input: Option<Value>,
+}
+
+impl EvalOuterIntersect {
+    pub(crate) fn new(setq: SetQuantifier) -> Self {
+        EvalOuterIntersect {
+            setq,
+            l_input: None,
+            r_input: None,
+        }
+    }
+}
+
+impl Evaluable for EvalOuterIntersect {
+    fn evaluate(&mut self, _ctx: &dyn EvalContext) -> Value {
+        let lhs = self.l_input.take().unwrap_or(Missing);
+        let rhs = self.r_input.take().unwrap_or(Missing);
+
+        match self.setq {
+            SetQuantifier::All => {
+                let mut result: Vec<Value> = Vec::new();
+                let mut lhs_multiplicities: HashMap<Value, usize> = HashMap::new();
+                coerce_to_bag(lhs).into_iter().for_each(|elem| {
+                    lhs_multiplicities
+                        .entry(elem)
+                        .and_modify(|m| *m += 1)
+                        .or_insert(1);
+                });
+                coerce_to_bag(rhs).into_iter().for_each(|elem| {
+                    if lhs_multiplicities.contains_key(&elem) {
+                        *lhs_multiplicities.get_mut(&elem).unwrap() += 1;
```

Comment:
I think this is wrong

Reply:
Yeah, not sure what I was originally doing there. Corrected in https://github.com/partiql/partiql-lang-rust/pull/400/commits/d1556dc1575fb23cd88d20f38754fab4a60adf56. Also added a new outer intersect test in https://github.com/partiql/partiql-tests/pull/101 that would have caught that mistake.